documentation/asciidoc/topics/ref_server_configuration.adoc

```diff
@@ -0,0 +1,34 @@
+| `deploy.container.resources.limits.cpu` | Defines the CPU limit, in CPU units, for each {brandname} pod. | `500m`
+| `deploy.container.resources.limits.memory` | Defines the memory limit, in bytes, for each {brandname} pod.|`512Mi`
+|`deploy.container.resources.requests.cpu`|Specifies the maximum CPU requests, in CPU units, for each {brandname} pod.|`500m`
+| `deploy.container.resources.requests.memory` | Specifies the maximum memory requests, in bytes, for each {brandname} pod. | `512Mi`
+| `deploy.security.secretName` | Specifies the name of a secret that creates credentials and configures security authorization. If you provide a security secret then `deploy.security.batch` does not take effect.|`""`
+| `deploy.security.batch` | Provides a batch file for the {brandname} CLI to create credentials and configure security authorization. The CLI runs the batch file before server startup. |`""`
+| `deploy.expose.type` | Specifies the service that exposes Hot Rod and REST endpoints outside the Kubernetes cluster and allows network access on the network. Set an empty value (`""`) if you do not want to expose {brandname} on the network. Valid options: [`""`, `"Route"`, `"LoadBalancer"`, `"NodePort"`]. | `Route`
+| `deploy.expose.nodePort` | Specifies a network port for node port services within the default range of 30000 to 32767. If you do not specify a port, the platform selects an available one. | `0`
+| `deploy.expose.host` | Specifies the hostname where the Ingress is exposed, if required. | `""`
+| `deploy.expose.annotations` | Adds annotations to the service that exposes {brandname} on the network. | `{}`
+|`deploy.logging.categories`| Specifies the FQN of a {brandname} package for which you want to configure logging. |`{}`
+|`deploy.logging.levels`| Specifies the level which {brandname} logs messages for a package. | `{}`
+| `deploy.resourceLabels` | Adds labels to {brandname} resources such as pods and services. | `{}`
+| `deploy.makeDataDirWritable` | Allows write access to the data directory for each {brandname} Server node. Setting the value to `true` creates an initContainer that runs `chmod -R` on the `/opt/infinispan/server/data` directory and changes its permissions.| `false`
+| `deploy.nameOverride` | Overrides the chart release name in the values file. Configure a name for the created resources only if you need it to be different to the Helm Chart release name. | Helm Chart release name
+| `deploy.infinispan` | {brandname} Server configuration. You should not change the default socket bindings or the security realm and endpoints named "metrics". Modifying these default properties can result in unexpected behavior and loss of service. | -
```

Comment:
`deploy.infinispan` is the main value to focus on in this reference topic imo. The other `deploy.*` values are documented and described in the README. We probably don't want to duplicate them here.

What if we have something like:

```
`deploy.infinispan`:: Configures cache managers and {brandname} Server endpoints, interfaces, socket bindings, and security.

[discrete]
== Cache manager configuration values

  infinispan:
    cacheContainer:
      name: default
      security:
        authorization: null
      transport:
        cluster: ${infinispan.cluster.name:cluster}
        node-name: ${infinispan.node.name:}
        stack: kubernetes

[discrete]
== Endpoint configuration values

  infinispan:
    server:
      endpoints:
      - securityRealm: default
        socketBinding: default
      - connectors:
          rest:
            restConnector:
              authentication:
                mechanisms: BASIC
        securityRealm: metrics
        socketBinding: metrics
```

Reply:
I'm kind of just throwing that in here but the idea is to break it down by section and provide an example with some details. I don't think we need tables that describe every one of those configuration parameters because they're common to Infinispan and we cover them in either the configuration guide or the server guide. We can just add links to those docs.

What I think we want to achieve is to provide the extra details for configuring clusters with `deploy.infinispan` values without users having to go to https://github.com/infinispan/infinispan-helm-charts/blob/main/values.yaml#L50 

The README mentions `deploy.infinispan` but not what can go in there, if that makes sense.